pkg/controller/acicontainersoperator.go

```diff
@@ -684,3 +827,235 @@ func (c *Controller) handleDaemonsetDelete(obj interface{}) bool {
+	if c.RoutesClient == nil {
+		log.Info("Route client is nil")
+		return nil
+	}
+	routes, err := c.RoutesClient.RouteV1().Routes(metav1.NamespaceAll).List(options)
+	if err != nil {
+		return err
+	}
+	var nodeAddress []string
+	nodeAddress, err = c.getNodeAddress()
+	if err != nil {
+		return err
+	}
+	log.Info("NodeAddress: ", nodeAddress)
+	// compute the dns servers info
+	var servers []operatorv1.Server
+	c.indexMutex.Lock()
+	for _, route := range routes.Items {
+		var server operatorv1.Server
+		key := route.ObjectMeta.Namespace + "/" + route.ObjectMeta.Name
+		server.Name = key
+		server.Zones = append(server.Zones, route.Spec.Host)
+		server.ForwardPlugin.Upstreams = nodeAddress
+		servers = append(servers, server)
+		c.routes[key] = true
+	}
+	c.indexMutex.Unlock()
+	update := false
+	if len(dnsInfo.Spec.Servers) == 0 {
+		dnsInfo.Spec.Servers = servers
+		update = true
+	} else {
+		if !reflect.DeepEqual(dnsInfo.Spec.Servers, servers) {
+			dnsInfo.Spec.Servers = servers
+			update = true
+		}
+	}
+	if update {
+		err = c.DnsOperatorClient.Update(context.TODO(), dnsInfo)
+		if err != nil {
+			return err
+		}
+	}
+	log.Infof("dnsInfo: %+v", dnsInfo)
+	return nil
+}
+
+func (c *Controller) getNodeAddress() ([]string, error) {
+	var options metav1.ListOptions
+	nodelist, err := c.K8s_Clientset.CoreV1().Nodes().List(options)
+	if err != nil {
+		log.Info(err)
+		return []string{}, err
+	}
+	var nodeAddress []string
+	for _, node := range nodelist.Items {
+		if node.DeletionTimestamp != nil {
+			continue
+		}
+		if _, ok := node.ObjectMeta.Labels["node-role.kubernetes.io/master"]; ok {
+			continue
+		}
+		address := node.Status.Addresses
+		for _, val := range address {
+			if val.Type == v1.NodeInternalIP {
+				nodeAddress = append(nodeAddress, val.Address)
+			}
+		}
+	}
+	return nodeAddress, nil
+}
+
+func (c *Controller) getDnsInfo() (*operatorv1.DNS, error) {
+	dnsInfo := &operatorv1.DNS{
+		TypeMeta:   metav1.TypeMeta{APIVersion: operatorv1.GroupVersion.String(), Kind: "DNS"},
+		ObjectMeta: metav1.ObjectMeta{Name: "default"},
+	}
+	err := c.DnsOperatorClient.Get(context.TODO(), types.NamespacedName{
+		Name: "default"}, dnsInfo)
+	if err != nil {
+		log.Info(err)
+		return nil, err
+	}
+	return dnsInfo, nil
+}
+
+// it reads all the node ip address.
+// updates if there is any changes in the address computed
+func (c *Controller) updateDnsOperatorSpec() bool {
+	if c.DnsOperatorClient == nil {
+		return false
+	}
+	dnsInfo, err := c.getDnsInfo()
+	if err != nil {
+		return true
+	}
+	if len(dnsInfo.Spec.Servers) == 0 {
+		return false
+	}
+	var nodeAddress []string
+	nodeAddress, err = c.getNodeAddress()
+	if err != nil {
+		log.Info(err)
+		return true
+	}
+	if !reflect.DeepEqual(dnsInfo.Spec.Servers[0].ForwardPlugin.Upstreams, nodeAddress) {
+		for _, server := range dnsInfo.Spec.Servers {
+			server.ForwardPlugin.Upstreams = nodeAddress
+		}
+		err = c.DnsOperatorClient.Update(context.TODO(), dnsInfo)
```

Comment:
If update is succesful it will be helpful to info log that it succeeded.

Reply:
Same here

Reply:
i have updated the log message